.github/workflows/preview.yml

```diff
@@ -0,0 +1,65 @@
+name: Deploy PR previews
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - closed
+
+concurrency: preview-${{ github.ref }}
+
+jobs:
+  deploy-preview:
+    runs-on: ubuntu-20.04
+    strategy:
+        matrix:
+          node-versions: [18.x]
+    env:
+      COMMIT_AUTHOR: Deploy Action
+      COMMIT_AUTHOR_EMAIL: action@github.com
+    steps:
+    - uses: actions/checkout@v2
+    - run: git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install prerequisites
+      run: pip install --upgrade wheel setuptools
+    - name: Install
+      run: pip install -e . -v
+    - name: fetch metadata
+      id: fetch_metadata
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: aiida-registry fetch
+    - name: Commit json file
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        mv plugins_metadata.json aiida-registry-app/src/
+        git config --local user.email "actions@github.com"
+        git config --local user.name "GitHub Actions"
+        git add aiida-registry-app/src/plugins_metadata.json
+        git commit -m "Add plugins data"
+        git push -f origin HEAD:gsoc
```

Comment:
Hi @AhmedBasem20, I think this might be the catch? You make pull request to `gsoc` branch, so before it merged, it may not be able to push to that branch.

Reply:
Meanwhile, I don't understand why you need to commit json file for the CI. I think Deploy preview only need file to show but not need to push any file to the repository. Am I missing something?

Reply:
Hi @unkcpz, I agree this might be the case, maybe the preview.yml needed to be merged in a separate PR before this one.
What do you think?

Reply:
I may try this on another branch on my fork first; create an equivalent branch to `gsoc`, push `preview.yml` to the branch, and open a pull request to my fork. If this works I think will work on the original repository too.

Reply:
Yes, I agree. It is better to be a separate PR. And we can rebase this one after merged.

Reply:
Yes, I agree. It is better to be a separate PR. And we can rebase this one after merged.

Reply:
Yes, I agree. It is better to be a separate PR. And we can rebase this one after merged.

Reply:
> Meanwhile, I don't understand why you need to commit json file for the CI. I think Deploy preview only need file to show but not need to push any file to the repository. Am I missing something?

This is correct! Why I'm doing all this at all!😄
I'll remove this step and see.

Reply:
We do eventually want to commit it to git (e.g. in order to look at historical data of the registry and, say, changes in some of the metrics over time).
however we can do this when we split up the "fetch data" and "build website" jobs (and the commit would be in the "fetch data" job).